### ISSUE TITLE
Stopping PHP-FPM: no child processes

### ISSUE THREAD

**Issue by RafaelKr** [symfony-cli/symfony-cli#412]
> Symfony CLI version 5.8.1 (c) 2021-2024 Fabien Potencier (2024-01-09T21:15:16Z - stable)

Recently my project suddenly stops being served. When running `symfony server:stop` I get the following error:

```
Stopping PHP-FPM: no child processes
Stopping Web Server

                                                                                                                        
 [OK] Stopped 2 process(es) successfully
```

![image](https://github.com/symfony-cli/symfony-cli/assets/14234815/0bef74e9-fba8-4895-aa75-5d46af676716)

Starting the server with `symfony server:start -d` works, but only for a limited amount of time.

Is there any way to provide you with more debug information?

**Comment by raneomik** [comment]
Hello, I have similar issue since the 5.8.1 update, not with php-fpm but with php : 

![image](https://github.com/symfony-cli/symfony-cli/assets/19893310/766e23ea-a881-4afb-b1fb-3942a873c945)

> Starting the server with symfony server:start -d works, but only for a limited amount of time.

I reprocude this issue when I work on several projects and start another php server, it says `[OK] Web server listening` but when I try to browse the web app, it goes in an inifinite loop.. On refresh, `server:log` shows this : 

![image](https://github.com/symfony-cli/symfony-cli/assets/19893310/cd859a02-b1b8-40bc-af2b-b83beb163bc5)

Thank you !

**Comment by sarim** [comment]
I'm also seeing this. This is because of 9eabae568c4cd2a0dbffde3bdee824caca203060 . The error  **ECHILD** is coming from syscall.Wait4. The logic implemented in that commit looks flawed to me.

Ref: 
- https://man7.org/linux/man-pages/man2/wait4.2.html
- https://man7.org/linux/man-pages/man2/waitpid.2.html
- https://pkg.go.dev/os#Process.Wait

> ECHILD (for waitpid() or waitid()) The process specified by pid
              (waitpid()) or idtype and id (waitid()) does not exist or
              is not a child of the calling process.

> On most operating systems, the Process must be a child of the current process or an error will be returned.

As described in the references, A process can't call **wait** on a pid that is not its child. Here `symfony server:start` command is its own command. Then we are running `symfony server:stop` as another command, it can't call wait on processes started by another process.

Another issue I noticed is after 5 second it returns with an error. So the last `syscall.Kill(-pgid, syscall.SIGKILL)` is never send. That leaves the processes running behind, but pid file will be cleaned up.

Maybe the behavior is different in macos? It even be different in musl vs glibc (as hinted in linux man page?). So a solution relying heavily on syscalls' might not be a good idea.

It should be based on pidfile's WaitForExit function. handled some edge cases quite well.
https://github.com/symfony-cli/symfony-cli/blob/e4dd0651f978e258436e35fab04724de0beff939/local/pid/pidfile.go#L113


Also in local_server_stop.go it is already doing SIGINT+waiting for main process, this could be modified to do the SIGINT+waiting-timeout for other worker processes. 
https://github.com/symfony-cli/symfony-cli/blob/e4dd0651f978e258436e35fab04724de0beff939/commands/local_server_stop.go#L56-L61

Also there's no SIGINT in windows.

Also the code path for pressing ctrl+c on `symfony server:start` and running `symfony server:stop` has become different. That'll just keep causing problems.

I suggest reverting the commit for now as hotfix, and reworking it later taking more time and taking all things into consideration. ATTN: @fabpot @Nihilus118

**Comment by fabpot** [comment]
Thank you for the detailed analysis.

> I suggest reverting the commit for now as hotfix, and reworking it later taking more time and taking all things into consideration. ATTN: @fabpot @Nihilus118

Reverted for now

**Comment by RafaelKr** [comment]
Thanks, that fixed it.
